:bug: Fix touchpad swipe back/forward #4246

dfelinto opened the pull request:
### Related Ticket

Github: #4246
Taiga: https://tree.taiga.io/project/penpot/issue/7454

### Summary

This is intentionally a fix for *most* of the cases. As commented on the code, users may still run into the original issue in some corner cases.

In my tests I need to go from the workspace to the dashboard and back very quickly for this to happen, and it doesn't even happen every time. I think having the ability to still use gestures on the dashboard justifies this quirk.

### Steps to reproduce

1. Go to the dashboard.
2. Open a document.
3. Use the trackpad to pan around.

The browser should *not* navigate back to the dashboard.

_Tested on MacOS + Google Chrome._

### Checklist

- [x] Choose the correct target branch; use `develop` by default.
- [x] Provide a brief summary of the changes introduced.
- [x] Add a detailed explanation of how to reproduce the issue and/or verify the fix, if applicable.
- [ ] ~~Include screenshots or videos, if applicable.~~
- [x] Add or modify existing integration tests in case of bugs or new features, if applicable. ¹
- [x] Check CI passes successfully.
- [x] Update the `CHANGES.md` file, referencing the related GitHub issue, if applicable.

¹ The following tests are failing in the `develop` branch, even without my patch:
```
[default] › playwright/ui/specs/subscriptions-dashboard.spec.js:85:3 › Subscriptions: team members and invitations › Team settings has susbscription name and no manage subscription link when is member 
[default] › playwright/ui/specs/variants.spec.js:114:1 › User duplicates a variant container 
```

niwinz commented on frontend/src/app/main/ui.cljs:
```diff
@@ -353,6 +353,15 @@
+  ;; In theory We could disable this only for the workspace. However gets too unreliable.
+  ;; It is better to be safe and disable for the dashboard as well.
+  (set! (.. js/document -documentElement -style -overscrollBehaviorX) "none")
+  (set! (.. js/document -body -style -overscrollBehaviorX) "none"))
```
I think we should name it `prevent-browser-gesture-navigation!`  (removing the `use-` prefix because it is not a react hook and right now use- prefix indicates a react hook)

And move it under `app.util.dom` because there we have all the dom manipulation helpers.

Commit:
:bug: Fix touchpad swipe back/forward #4246

This prevents the browser to take over the trackbad swipe gesture both
for the dashboard and the workspace.

At an early attempt I did get the code to work only for the workspace,
but it is too unreliable and I could every now and then get it to misbehave.

I believe it is better to be safe and always prevent the browser from
going back/forth, regardless of workspace/dashboard.

Signed-off-by: Dalai Felinto <dalai@blender.org>
